### ISSUE TITLE
License extension: For ethical use only

### ISSUE THREAD

**Issue by aschrijver** [Widdershin/programmers-oath#90]
Thanks for this great initiative. I signed up :)

I would like to point to a discussion I started on the Discourse forum of the recently founded [Center for Humane Technology](https://humanetech.com) on the possibility to create a **_For Ethical Use Only_** OSS license type, or extension. Very much related IMO to this oath, but more formal (but maybe hard to realize).

See the discussion: [Idea - Extending OSS licenses: For Ethical Use Only](https://community.humanetech.com/t/idea-extending-oss-licenses-for-ethical-use-only/544)

Would like to hear your opinions on the feasibility of this idea..

PS I'll cross-post a link to this repo on the same thread and add you to the [Awesome Humane Tech](https://github.com/engagingspaces/awesome-humane-tech) list

**Comment by mo-g** [comment]
Marking this `invalid` because it's not actually an issue with the oath itself - but I'll leave it open, because it's definitely worth sharing. To answer some of your questions from the original discussion - yes, it has been done.

Short and sweet: [the JSON licence](https://spdx.org/licenses/JSON).

There were earlier and more complex alternatives, including no-nukes licences, and licences that banned use by military or certain other options.

As I'm sure you've discovered while looking into this concept, those licences are explicitly non-free, and depending on definition may not be considered 'open source'. There is no way to work around those limitations, because the definitions explicitly prohibit discrimination of use.

It would be good to see a more comprehensively thought out creation - perhaps as a 'patch' that could be applied to existing free licences? Definitely worthy of a github repo. One risk; an Oath is a humanistic agreement and any human can meaningfully contribute. A licence is a legal agreement and it would be unwise to use a licence not drafted or inspected by some very well qualified and probably expensive lawyers.

**Comment by aschrijver** [comment]
Thx for your response @mo-g and leaving the issue open :)

> Short and sweet: the JSON licence.
>
> There were earlier and more complex alternatives, including no-nukes licences, and licences that banned use by military or certain other options.

The JSON license is _really short and sweet_ only stating: **The Software shall be used for Good, not Evil.**
I was thinking of including something more elaborate, and - indeed - more complex and elaborate. I am hoping The Center for Humane Technology and others can offer a helping hand.. there are already a number of lawyers that are member.

With your permission I would like to copy your comment to the [discussion thread](https://community.humanetech.com/t/idea-extending-oss-licenses-for-ethical-use-only/544) over there (and refer to your github account in the quote). Can I do that? Or maybe you'd like to join that community yourself?

**Comment by mo-g** [comment]
I have enough accounts for things, forgive me for not signing up to more services unless I know I'm going to be a heavy used. :wink: More than happy for you to quote me on this regard, as long as you don't twist my words into something evil. :stuck_out_tongue:

**Comment by aschrijver** [comment]
Ha ha, don't worry @mo-g I won't, I won't :smile: Thx!

PS. One relevant pointer to oaths and humane tech:

----
[Tristan Harris] .. is genuinely afraid — of Big Tech ruining our lives. “The Internet was supposed to be this exciting potential to do all this good,” he says mournfully, sitting on one of those comical lawn chairs, here in the heart of Internet optimism. Not only has the Web failed to realize its full potential but it, and the people building it, also might be doing us quite a lot of harm.

http://www.ozy.com/fast-forward/special-series-what-if-designers-took-a-hippocratic-oath/64873

**Comment by bortzmeyer** [comment]
Clearly a bad idea if there is no definition of "ethical". Is the use of my software by the YPG https://en.wikipedia.org/wiki/People%27s_Protection_Units ethical? Some pacifists will say no, no military use can be ethical, other people may consider that participating in a just war is ethical. See the discussion in #25

**Comment by aschrijver** [comment]
@bortzmeyer I agree. If stating 'ethical' it should not be in a single sentence, and it should probably also not be a 'full' definition of what constitutes ethical (this would be by definition a personal personal perspective, and maybe even impossible to define in a legal sense).

Maybe some very general definition that widely applies, is not too restrictive and/or some added and obvious examples of non-ethical uses that are explicitly forbidden.

Regarding military use: It could have variations of the license, just like with creativecommons (FOSS), creativecommons minus commercial use (non-free), etc.

Note that any inclusions that make the license non-free (which will quickly be the case) will lead to the same issue + discussions the JSON license had. But as a software developer you should be free to use the license you want. Other software developers should take care not to use it when they do not agree. A delicate topic, I know.. I am just philosophizing here, and no expert in this :)

**Comment by mo-g** [comment]
> Regarding military use: It could have variations of the license, just like with creativecommons (FOSS), creativecommons minus commercial use (non-free), etc.

Interesting solution. There's also the GPL approach; GPLv3 was simply a replacement for v2 that added 'ethical exemptions' to reduce the problems of e-waste and 'split ownership' of devices. Some people adopt it, some don't.

Your point about JSON is well made: Everyone is free to not be free. The point of licences being free is a purely moral one: the idea that no one should have exclusive ownership of a piece of math. For other people, their moral perspective may differ: for them, their work went into the creation or discovery of that piece of math, and if it is used for evil they have committed evil themselves. To prevent that in a way, is also exercising freedom of speech.
